src/api/routes/rosetta/construction.ts

```diff
@@ -297,5 +322,147 @@ export function createRosettaConstructionRouter(db: DataStore): RouterWithAsync
     }
   });
 
+  //construction/payloads endpoint
+  router.postAsync('/payloads', async (req, res) => {
+    const valid: ValidSchema = await rosettaValidateRequest(req.originalUrl, req.body);
+    if (!valid.valid) {
+      res.status(400).json(makeRosettaError(valid));
+      return;
+    }
+
+    const options = getOptionsFromOperations(req.body.operations);
+    if (options == null) {
+      res.status(400).json(RosettaErrors.invalidOperation);
+      return;
+    }
+
+    const amount = options.amount;
+    if (!amount) {
+      res.status(400).json(RosettaErrors.invalidAmount);
+      return;
+    }
+
+    const fees = options.fee;
+    if (!fees) {
+      res.status(400).json(RosettaErrors.invalidFees);
+      return;
+    }
+
+    const publicKeys: RosettaPublicKey[] = req.body.public_keys;
+    if (!publicKeys) {
+      res.status(400).json(RosettaErrors.emptyPublicKey);
+      return;
+    }
+
+    if (publicKeys[0].curve_type !== 'secp256k1') {
+      res.status(400).json(RosettaErrors.invalidCurveType);
+      return;
+    }
+
+    const recipientAddress = options.token_transfer_recipient_address
+      ? options.token_transfer_recipient_address
+      : '';
+    const senderAddress = options.sender_address ? options.sender_address : '';
+
+    const accountInfo = await new StacksCoreRpcClient().getAccount(senderAddress);
```

Comment:
What happens when the above `sender_address` or `recipient_address` are empty/undefined? Seems like this would be an error condition?